File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/output/project/src/surveycrawler/ground_map_sonar.py
from surveycrawler.firm_sand import FirmSand
from surveycrawler.sonar import Sonar


class GroundMapSonar(Sonar):

    # instance creation

    @classmethod
    def with_grounds_at(cls, a_ground_by_position):
        return cls(a_ground_by_position)

    # initialization

    def __init__(self, a_ground_by_position):
        self._grounds = a_ground_by_position

    # ground detection

    def ground_at(self, a_position):
        return self._grounds.get(a_position, FirmSand())

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-212555-19812/output/project/src/main/java/surveycrawler/GroundMapSonar.java
package surveycrawler;

import java.util.HashMap;
import java.util.Map;

public class GroundMapSonar implements Sonar {

    private final Map<Point, Ground> groundsByPosition = new HashMap<>();
    private final Ground defaultGround = new FirmSand();

    // instance creation

    public static GroundMapSonar empty() {
        return new GroundMapSonar();
    }

    private GroundMapSonar() {
    }

    // configuring

    public GroundMapSonar with(Point aPosition, Ground aGround) {
        groundsByPosition.put(aPosition, aGround);
        return this;
    }

    // sensing

    @Override
    public Ground groundAt(Point aPosition) {
        return groundsByPosition.getOrDefault(aPosition, defaultGround);
    }
}
